src/api/removeOrphanPkgs.ts

```diff
@@ -47,6 +50,32 @@ export default async function removeOrphanPkgs (
 
   if (!opts.dryRun) {
     if (notDependents.length) {
+
+      if (opts.shamefullyFlatten && opts.oldShrinkwrap.packages) {
+        const packages = opts.oldShrinkwrap.packages
+        await Promise.all(notDependents.map(async notDependent => {
+          const pkgShr = packages[dp.relative(opts.oldShrinkwrap.registry, notDependent)]
+          const pkgInfo = getPkgInfoFromShr(notDependent, pkgShr)
+          const rootLink = path.join(rootModules, pkgInfo.name)
+          // rootLink might not exist anymore because it might have been delete earlier
+          if (await exists(rootLink)) {
```

Comment:
this won't always work. A package can be aliased and in that case the symlink in the root will be named differently.

The `.name` in the shrinkwrap will always be real name.

A better solution might be to collect all the symlink targets in node_modules and compare them to the notDependents.